### ISSUE TITLE
.Text returns empty-string

### ISSUE THREAD

**Issue by HalosGhost** [SeleniumHQ/selenium#8502]
## 🐛 Bug Report

`WebElement.Text` returns `""`. In my very limited testing, it looks like this _may_ be because the title tag of my website (the site I'm testing against) contains slashes.

## To Reproduce

Minimal, complete example:

Detailed steps to reproduce the behavior:

* Add the `.cs` file given below in the test script section to a clean MSTest project, and install the required nuget packages.
* Run the tests

## Expected behavior

* That all tests should pass

## Actual behavior

* the test on line 69 (testing against ``.Text``) fails (commenting out line 69 leaves all tests passing)
* the use of ``.Text`` with the `h1` (which does not contain slashes) returns the expected string
* the use of the less-standard `innerHTML` attribute returns the expected text

## Test script or set of commands reproducing this issue

```csharp
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Firefox;
using System;

// cf. https://dzone.com/articles/developer-tips-c-selenium-with-mstest-basics

namespace selenium
{
    public enum BrowserDriver
    {
        Chrome,
        Firefox
    }

    internal static class WebDriverFactory
    {
        public static IWebDriver Create(BrowserDriver driver, string seleniumWebDriversPath)
        {
            return driver switch
            {
                BrowserDriver.Chrome => new ChromeDriver(seleniumWebDriversPath),
                BrowserDriver.Firefox => new FirefoxDriver(seleniumWebDriversPath),
                _ => throw new ArgumentOutOfRangeException(nameof(driver), driver, null),
            };
        }
    }

    [TestClass]
    public class PageLoadsTest
    {
        public string DriverPath { get; }
        public string URL { get; }

        public PageLoadsTest()
        {
            DriverPath = AppDomain.CurrentDomain.SetupInformation.ApplicationBase;
            URL = "https://halosgho.st";
        }

        [TestMethod]
        [DataRow(BrowserDriver.Chrome)]
        [DataRow(BrowserDriver.Firefox)]
        public void PageLoads(BrowserDriver engine)
        {
            using var driver = WebDriverFactory.Create(engine, DriverPath);
            driver.Navigate().GoToUrl(URL);

            var title = driver.FindElement(By.TagName("title"));
            Assert.AreEqual("/home/halosghost", title.GetAttribute("innerHTML"));
            Assert.AreEqual("/home/halosghost", title.Text);

            var heading = driver.FindElement(By.TagName("h1"));
            Assert.AreEqual("Sam Stuewe (halosghost)", heading.GetAttribute("innerHTML"));
            Assert.AreEqual("Sam Stuewe (halosghost)", heading.Text);

            driver.Quit();
        }
    }
}
```

## Environment

OS: Windows 10
Browser: latest versions of Chromium and Firefox
Browser Driver version: latest versions of geckodriver and chromedriver
Language Bindings version: C# 3.141.0

**Comment by HalosGhost** [comment]
Additionally, `.GetAttribute("innerText")` (which is, hypothetically, what `.Text` should be retrieving) also returns the expected result. Further, another page I tested (whose `<title>` contains `|`s (but no slashes)) experiences this same behavior.

*Edit*: I did not actually test with IE or Edge (because I do not have them readily available at the moment). However, given that this affects both Chrome and Firefox, I suspect the issue is a bit deeper.

*Edit 2*: a further test indicates that `IWebDriver.Title` returns the expected text.

*Edit 3*: I have shortened the test code just to make sure that some of my ancillary inclusions do not affect the result.

**Comment by AutomatedTester** [comment]
This is by design. 

`.Text` returns the visible text as described in the [spec](https://w3c.github.io/webdriver/#get-element-text).

Since `<title>` is not a visible element on the page calling `Text` on it will return an empty string. I suggest doing

I suggest using `innerText` as you did in your comment.

**Comment by HalosGhost** [comment]
It may benefit future new-comers to the project if the documentation reflects that.

**Comment by AutomatedTester** [comment]
fixed in ebadcd05a247e4e041b7e73119fb7367696e8274

**Comment by HalosGhost** [comment]
@AutomatedTester, I appreciate that change immensely. I think it is much more honest about the functionality that can be expected. The only thing I might recommend is using “e.g.,” instead of “i.e.,” since `<title>` isn't hidden by CSS, it's just not rendered on-page (so, “hidden by CSS” is an example of an exception, not the sole, specific exception). But, I recognize that might be nit-picking. :)
